packages/core-transaction-pool/src/connection.ts

```diff
@@ -53,11 +53,8 @@ export class Connection implements TransactionPool.IConnection {
         }
 
         this.emitter.once("internal.stateBuilder.finished", async () => {
-            const validTransactions = await this.validateTransactions(transactionsFromDisk);
-            transactionsFromDisk = transactionsFromDisk.filter(transaction =>
-                validTransactions.includes(transaction.serialized.toString("hex")),
-            );
-
+            // Remove from the pool invalid entries found in `transactionsFromDisk`.
+            await this.validateTransactions(transactionsFromDisk);
```

Comment:
Notice: we don't need the `transactionsFromDisk` variable to be updated here. This code is executed just once.